### DISCUSSION TITLE
Is there any way to use DataList Value "Name" property in Name Templates (for example, in a Block List Editor)?

### DISCUSSION THREAD

**Discussion by hfloyd** [leekelleher/umbraco-contentment#302] [Question] [answered]
I have a DataList that returns Umbraco Content Nodes as the "pickable" values:

![image](https://user-images.githubusercontent.com/1252593/215615895-0c3817e8-8d19-44a1-929a-3d5828b67a76.png)

I know that when picked, the data which will be stored is the UDI for the node. I was wondering, though, if I am using this picker as a property editor on a Content Block, and I want to enable the dynamic "Label" field, is there a way to access the "Name" rather than the "Value"?

If I set it to something like this:

`{{!!SelectedContent  ? SelectedContent :' Library Content'}}`

It renders the UDI value:
![image](https://user-images.githubusercontent.com/1252593/215616373-d15dfbc9-a9fb-404d-a79f-1af3e981c8b8.png)

If I try something like this:
`{{!!SelectedContent  ? SelectedContent.name :' Library Content'}}`
or
`{{!!SelectedContent  ? SelectedContent["name"] :' Library Content'}}`

It renders nothing, indicating that it recognizes there IS a value there, but there is nothing for "name":
![image](https://user-images.githubusercontent.com/1252593/215616678-800ebd4e-7f92-47d0-8ff2-030657d7d1f0.png)

If I try something like this:
`{{!!SelectedContent  ? (SelectedContent |  ncNodeName) :' Library Content'}}`

I get console errors:
![image](https://user-images.githubusercontent.com/1252593/215617478-f790194e-da15-450b-9d34-ff8fed448416.png)

```
angular.js:15697 TypeError: input.split is not a function
    at nodeNameFilter (umbraco.filters.min.js?d=638107170200000000:1:1979)
    at fn (eval at compile (angular.js:16548:15), <anonymous>:4:466)
    at angular.js:17559:31
    at m.$digest (angular.js:19262:34)
    at angular.js:19562:15
    at Yg.completeTask (angular.js:21403:7)
    at angular.js:6879:7
```
**Is there a way to access the DataList properties other than "Value" for a Name Template?**

**Comment by leekelleher** [comment, marked as answer]
Yes and no.

No, in that only the "value" is available at the point of rendering the Block item.

Yes, in that for Umbraco content, I've added an undocumented custom AngularJS filter for this, instead of `ncNodeName`, try `lkNodeName`.

**Reply by hfloyd** [reply]
Ah, we were both responding at the same time! I see that your custom name filter also work the same way.

`(SelectedContent[0] |  lkNodeName)`

**Comment by hfloyd** [comment]
Ah! I figured it out.

My DataList allows multiple selections, so if I use this:

`(SelectedContent[0] |  ncNodeName)`

It works!

Since in this instance, I only want to allow the Editor to make a single selection, I copied the DataList and made a "single-select" version. However, the same Name Template (with the [0]) works.

**Reply by leekelleher** [reply]
Give the `lkNodeName` filter a try, it supports multiple values.

**Reply by hfloyd** [reply]
Thanks, @leekelleher 🤗

**Comment by leekelleher** [comment]
A little background on the `lkNodeName` filter; ultimately it's a wrapper for `ncNodeName`.
If the value is single, then it'll pass it straight to `ncNodeName` - otherwise if it's an array, it'll loop over the values, passing each one to `ncNodeName`, then concatenating them with a comma. So all the inner logic/fetching is done within `ncNodeName`, rather than me re-inventing the wheel.

The source is here: https://github.com/leekelleher/umbraco-contentment/blob/4.4.1/src/Umbraco.Community.Contentment/DataEditors/_/_filters.js#L6-L19

**Reply by hfloyd** [reply]
Brilliant stuff, @leekelleher ! Seems the UmbracoCore ncNodeName should be as clever! 👏🏻
